src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -153,12 +153,27 @@ export async function migrationStateActionMachine({
       logger.error(
         logMessagePrefix + `[${e.body?.error?.type}]: ${e.body?.error?.reason ?? e.message}`
       );
+      dumpExecutionLog(logger, logMessagePrefix, executionLog);
+      throw new Error(
+        `Unable to complete saved object migrations for the [${
+          initialState.indexPrefix
+        }] index. Please check the health of your Elasticsearch cluster and try again. Error: [${
+          e.body?.error?.type
+        }]: ${e.body?.error?.reason ?? e.message}`
+      );
     } else {
       logger.error(e);
+
+      dumpExecutionLog(logger, logMessagePrefix, executionLog);
+      if (e.message.startsWith('Unable to migrate the corrupt saved object document')) {
```

Comment:
NIT: I would use error sub-classing instead of relying on an error message for the handling behavior change.

Reply:
yeah I was being lazy :(